esmf_regrid/tests/unit/experimental/unstructured_scheme/test_regrid_unstructured_to_rectilinear.py

```diff
@@ -55,3 +57,51 @@ def _add_metadata(cube):
     # Check metadata and scalar coords.
     expected_cube.data = result.data
     assert expected_cube == result
+
+
+def test_multidim_cubes():
+    """
+    Test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_unstructured_to_rectilinear`.
+
+    Tests with multidimensional cubes. The source cube contains
+    coordinates on the dimensions before and after the mesh dimension.
+    """
+    mesh = _full_mesh()
+    mesh_length = mesh.connectivity(contains_face=True).shape[0]
+
+    h = 2
+    t = 3
+    height = DimCoord(np.arange(h), standard_name="height")
+    time = DimCoord(np.arange(t), standard_name="time")
+
+    src_data = np.empty([t, mesh_length, h])
+    src_data[:] = np.arange(t * h).reshape([t, h])[:, np.newaxis, :]
+    cube = Cube(src_data)
+    mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+    cube.add_aux_coord(mesh_coord_x, 1)
+    cube.add_aux_coord(mesh_coord_y, 1)
+    cube.add_dim_coord(time, 0)
+    cube.add_dim_coord(height, 2)
+
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+    result = regrid_unstructured_to_rectilinear(cube, tgt)
+
+    # Lenient check for data.
+    expected_data = np.empty([t, n_lats, n_lons, h])
+    expected_data[:] = np.arange(t * h).reshape(t, h)[:, np.newaxis, np.newaxis, :]
```

Comment:
What is the purpose of the empty -> fill with arange construction?

This test and the `MeshToGridESMFRegridder` test are very similar, is there a reason we're using different test data for each?  Should we have a test that compares both the object and the function return the same thing?

Reply:
The idea here is that each horizontal level is a unique constant. I do this by broadcasting an array filled with different numbers from `np.arange`.
The purpose here is to ensure that each horizontal level is mapped to the appropriate place in the final array. Having each level being constant makes this straightforward to check that this calculation is correct. I've changed the other test in `MeshToGridESMFRegridder` to be equivalent.